### DISCUSSION TITLE
Multiple namespaces and services accounts on a cluster.

### DISCUSSION THREAD

**Discussion by Victor-2021** [argoproj/argo-cd#5193] [Q&A]
Hi,

We are trying to manage multiple namespaces on a same cluster which ArgoCD would not have full access to. Enrolling only specific namespaces is supported with `--namespace` option, however in our case different namespaces have different service accounts which have access to them. In order words on the same cluster we have Namespace A and Service Account A that has access to it alone, and Namespace B and Service Account B that has access to it alone. We are looking to deploy to the specified namespaces on behalf of the associated service account. `Cluster add` command allows providing a service account for cluster, but I do not see a way to associate multiple service accounts with multiple namespaces.

Is there a mechanism for providing specific service accounts to be associated with specific namespaces on the same cluster? 

Thank you

**Comment by kshamajain99** [comment]
@Victor-2021 When providing service account via `cluster add --service-account`, you provide system namespace service account which has permissions to manage kubernetes resources accross all namespaces. Can you please elaborate your usecase as to why do you want to provide namespace specific service account?
